server/repository/src/db_diesel/changelog/changelog.rs

```diff
@@ -0,0 +1,199 @@
+use crate::{
+    diesel_macros::apply_equal_filter, DBType, EqualFilter, RepositoryError, StorageConnection,
+};
+use diesel::{helper_types::IntoBoxed, prelude::*};
+use std::convert::TryInto;
+use util::inline_init;
+
+use diesel_derive_enum::DbEnum;
+
+table! {
+    changelog (cursor) {
+        cursor -> BigInt,
+        table_name -> crate::db_diesel::changelog::ChangelogTableNameMapping,
+        record_id -> Text,
+        row_action -> crate::db_diesel::changelog::ChangelogActionMapping,
+        name_id -> Nullable<Text>,
+        store_id -> Nullable<Text>,
+    }
+}
+
+table! {
+    changelog_deduped (cursor) {
+        cursor -> BigInt,
+        table_name -> crate::db_diesel::changelog::ChangelogTableNameMapping,
+        record_id -> Text,
+        row_action -> crate::db_diesel::changelog::ChangelogActionMapping,
+        name_id -> Nullable<Text>,
+        store_id -> Nullable<Text>,
+    }
+}
+
+#[derive(DbEnum, Debug, Clone, PartialEq, Eq)]
+#[DbValueStyle = "SCREAMING_SNAKE_CASE"]
+pub enum ChangelogAction {
+    Upsert,
+    Delete,
+}
+
+#[derive(DbEnum, Debug, Clone, PartialEq, Eq)]
+#[DbValueStyle = "snake_case"]
+pub enum ChangelogTableName {
+    Number,
+    Location,
+    StockLine,
+    Name,
+    NameStoreJoin,
+    Invoice,
+    InvoiceLine,
+    Stocktake,
+    StocktakeLine,
+    Requisition,
+    RequisitionLine,
+}
+
+#[derive(Clone, Queryable, Debug, PartialEq, Insertable)]
```

Comment:
Added insertable, since it's easier to test some things against direct insert vs through trigger